Use PCDIR CMake option, if configured

Upstream install the server pkgconfig file into arch-independent directory
Reported to upstream as: https://jira.mariadb.org/browse/MDEV-14340

diff -Naupr mariadb-10.4.27.orig/support-files/CMakeLists.txt mariadb-10.4.27/support-files/CMakeLists.txt
--- mariadb-10.4.27.orig/support-files/CMakeLists.txt	2022-12-05 11:11:14.647655859 +0100
+++ mariadb-10.4.27/support-files/CMakeLists.txt	2022-12-05 11:12:03.789128070 +0100
@@ -93,7 +93,11 @@ IF(UNIX)
 
   CONFIGURE_FILE(mariadb.pc.in ${CMAKE_CURRENT_BINARY_DIR}/mariadb.pc @ONLY)
   CONFIGURE_FILE(rpm/server.cnf ${CMAKE_CURRENT_BINARY_DIR}/rpm/server.cnf @ONLY)
-  INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/mariadb.pc DESTINATION ${INSTALL_LIBDIR}/pkgconfig COMPONENT Development)
+  IF(INSTALL_PCDIR)
+    INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/mariadb.pc DESTINATION ${INSTALL_PCDIR} COMPONENT Development)
+  ELSE()
+    INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/mariadb.pc DESTINATION ${INSTALL_LIBDIR}/pkgconfig COMPONENT Development)
+  ENDIF()
 
   INSTALL(FILES mysql.m4 DESTINATION ${INSTALL_SHAREDIR}/aclocal COMPONENT Development)
   
